pkg/kafka/partition/offset_manager.go

```diff
@@ -57,53 +59,45 @@ func NewKafkaOffsetManager(
 
 	return newKafkaOffsetManager(
 		c,
-		cfg.Topic,
-		partitionID,
-		cfg.GetConsumerGroup(instanceID, partitionID),
+		cfg,
+		instanceID,
 		logger,
 	), nil
 }
 
 // newKafkaReader creates a new KafkaReader instance
 func newKafkaOffsetManager(
 	client *kgo.Client,
-	topic string,
-	partitionID int32,
-	consumerGroup string,
+	cfg kafka.Config,
+	instanceID string,
 	logger log.Logger,
 ) *KafkaOffsetManager {
 	return &KafkaOffsetManager{
-		client:        client,
-		topic:         topic,
-		partitionID:   partitionID,
-		consumerGroup: consumerGroup,
-		logger:        logger,
+		client:      client,
+		adminClient: kadm.NewClient(client),
+		cfg:         cfg,
+		instanceID:  instanceID,
```

Comment:
nit: should we just save the result of `r.cfg.GetConsumerGroup(r.instanceID)` as consumerGroup here instead of calling this method each time we need `ConsumerGroup()`?

Reply:
Yeah a nice optimization, will leave for followup.